src/main/java/io/managed/services/test/dataplane/FleetshardUtils.java

```diff
@@ -0,0 +1,71 @@
+package io.managed.services.test.dataplane;
+
+import io.fabric8.kubernetes.api.model.KubernetesResourceList;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.MixedOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.fabric8.openshift.client.OpenShiftClient;
+import io.managed.services.test.k8.managedkafka.v1alpha1.ManagedKafka;
+import io.managed.services.test.k8.managedkafka.v1alpha1.ManagedKafkaAgent;
+import lombok.extern.log4j.Log4j2;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Log4j2
+public class FleetshardUtils {
+
+    public static MixedOperation<ManagedKafka, KubernetesResourceList<ManagedKafka>, Resource<ManagedKafka>> managedKafka(KubernetesClient client) {
+        return client.resources(ManagedKafka.class);
+    }
+
+    public static MixedOperation<ManagedKafkaAgent, KubernetesResourceList<ManagedKafkaAgent>, Resource<ManagedKafkaAgent>> managedKafkaAgent(KubernetesClient client) {
+        return client.resources(ManagedKafkaAgent.class);
+    }
+
+    public static int getCountOfExistingGivenManagedKafkaCRType(OpenShiftClient oc, ManagedKafkaType mkType) {
+        return (int) FleetshardUtils.managedKafka(oc).inAnyNamespace().list().getItems()
+            .stream()
+            .filter(e -> mkType.toString().equals(e.getLabel("bf2.org/kafkaInstanceProfileType").orElse("notPresent")))
+            .count();
```

Comment:
```suggestion
        return listManagedKafka(oc,mkType).size();
```

We can re-use the `listManagedKafka` util here

Reply:
The method counts the number of MK CRs - `reserved`/`prewarmed` and `real`.

Reply:
**about counting with reserved deployment CR**:  yeah the test count with that i did not want to make super specialized for eliminating  by namespace/ or label. Tests count with it and i will rename the method to match this fact (that it actually counts CRs)

reusing method. Correct (i implemented them reversed order so i totally forgot).

Reply:
> about counting with reserved deployment CR: yeah the test count with that i did not want to make super specialized for eliminating by namespace/ or label. Tests count with it and i will rename the method to match this fact (that it actually counts CRs)

Might be worth it to explicit this in the javadoc